libpod/container_log.go

```diff
@@ -21,10 +21,31 @@ func init() {
 	logDrivers = append(logDrivers, define.KubernetesLogging, define.NoLogging, define.PassthroughLogging)
 }
 
+func cloneLogOpts(options *logs.LogOptions, newColor int64) *logs.LogOptions {
+	clone := logs.LogOptions{
+		Details:    options.Details,
+		Follow:     options.Follow,
+		Since:      options.Since,
+		Until:      options.Until,
+		Tail:       options.Tail,
+		Timestamps: options.Timestamps,
+		Colors:     options.Colors,
+		ColorId:    newColor,
+		Multi:      options.Multi,
+		WaitGroup:  options.WaitGroup,
+		UseName:    options.UseName,
+	}
+	return &clone
+}
+
 // Log is a runtime function that can read one or more container logs.
 func (r *Runtime) Log(ctx context.Context, containers []*Container, options *logs.LogOptions, logChannel chan *logs.LogLine) error {
-	for _, ctr := range containers {
-		if err := ctr.ReadLog(ctx, options, logChannel); err != nil {
+	for c, ctr := range containers {
+		clonedOpts := options
+		if options.Colors {
+			clonedOpts = cloneLogOpts(options, int64(c))
```

Comment:
could this pass in 7 or higher?  Do we need a color_max check somewhere?

Reply:
Currently, the modulo of the `colorID` of the container is taken as the color index for that container. So the eighth container in a pod would have the same color as the first, in a loop. Would this solution do?